monai/data/itk_torch_affine_matrix_bridge.py

```diff
@@ -0,0 +1,306 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, cast
+
+import numpy as np
+import torch
+
+from monai.config import NdarrayOrTensor
+from monai.data import ITKReader
+from monai.data.meta_tensor import MetaTensor
+from monai.transforms import Affine, EnsureChannelFirst
+from monai.utils import convert_to_dst_type, optional_import
+
+if TYPE_CHECKING:
+    import itk
+
+    has_itk = True
+else:
+    itk, has_itk = optional_import("itk")
+
+__all__ = [
+    "metatensor_to_array",
+    "image_to_metatensor",
+    "remove_border",
+    "itk_to_monai_affine",
+    "monai_to_itk_affine",
+    "create_itk_affine_from_parameters",
+    "itk_affine_resample",
+    "monai_affine_resample",
+]
+
+
+def metatensor_to_array(metatensor: MetaTensor):
+    metatensor = metatensor.squeeze()
+    metatensor = metatensor.permute(*torch.arange(metatensor.ndim - 1, -1, -1))
```

Comment:
the writer class also does this permute

https://github.com/Project-MONAI/MONAI/blob/50c3f328cb98e4f0c4e6a38a04a98badf0961c18/monai/data/image_writer.py#L507

so this shouldn't be done with the SaveImage to avoid permute twice...

Reply:
I don't quite understand what you mean with this. Should I remove this function and just use `ITKWriter.create_backend_obj()` instead?

Reply:
I think `metatensor_to_array` should be removed... the `meta_tensor.array` property returns the numpy array. any data format change/assumption shouldn't be hard-coded in this way, unless I miss something here cc @ntatsisk

Reply:
Looking back I think the naming of the function could be better. The idea is that you need to match the axis of the array the comes from `np.asarray(itk_image)`, and in order to do that you need a squeeze and an axis-swap. Feel free to remove it, but these two operations need to be done then inline before comparing to the array that comes from itk. That at least for testing purposes (no image saving there).